packages/framework/src/Framework/Features/Navigation/NavigationMenu.php

```diff
@@ -19,9 +19,7 @@
  */
 class NavigationMenu
 {
-    public Route $currentRoute;
-
-    public Collection $items;
+    protected Collection $items;
```

Comment:
Actually, since the `$items` property is a collection that is itself writeable, making it protected makes no effective change as you can still modify it using the getter.

Reply:
We could handle this by using the getter to create a new collection instance in case we do want to scope down the API and make it read-only (you can still push pages to the router). We'll see...